core/src/main/java/hudson/model/Job.java

```diff
@@ -1210,7 +1210,8 @@ public synchronized void doConfigSubmit(StaplerRequest req,
 
             String newName = req.getParameter("name");
             final ProjectNamingStrategy namingStrategy = Jenkins.getInstance().getProjectNamingStrategy();
-            if (newName != null && !newName.equals(name)) {
+            if (newName != null && !newName.trim().equals(name.trim())) {
```

Comment:
@kzantow as i see you are newly core contributor, do you have any argument about block of existed code i.e. "no, it works wonderful", "you are wrong, this is good logic" rather then discussing my comment?
@kzantow as core contributor please suggest something for this block of code.